### ISSUE TITLE
imrsh: "echo æøå" in history causes CPU spike and deadlock

### ISSUE THREAD

**Issue by kseistrup** [emersion/mrsh#147]
I have the following in `~/.local/share/imrsh/history`:

```txt
help
echo æøå
dir
ps
```

When I start a new `imrsh` instance and press _arrow up_ I will first see `$ ps▒` and `$ dir▒`, where `▒` is the cursor, then `$ echo æøå` where the cursor is located over the `e` in `echo`. Here the shell will become irresponsive and loadavg will climb to 100%. The process doesn't respond to `^C` — I have to kill the process in order to regain control.

----

I am using https://aur.archlinux.org/packages/imrsh-git to install `imrsh`.

* imrsh r28.da592a3
* libtickit 0.3.3 (and bzr r322)
* libtermkey 0.22

```sh
$ uname -a
Linux home 5.5.7-arch1-1 #1 SMP PREEMPT Sat, 29 Feb 2020 19:06:02 +0000 x86_64 GNU/Linux
$ echo $TERM
xterm-256color
$ echo $LANG
en_DK.UTF-8
$ echo $LC_ALL
en_DK.UTF-8
```

Seen in terminal emulators:

```sh
$ sakura --version
sakura version is 3.7.0
$ lilyterm --version
LilyTerm 0.9.9.4 (2020-02-08)
```

**Comment by emersion** [comment]
Can't reproduce with `mrsh` itself. I think UTF-8 support is still not done for imrsh.

**Comment by kseistrup** [comment]
It is correct that it doesn't happen in `mrsh`, only in `imrsh`. I suspect it is a problem with whatever `imrsh` is using for providing readline functionality.

**Comment by ddevault** [comment]
Please report future imrsh bugs to the appropriate tracker: https://todo.sr.ht/~sircmpwn/imrsh

**Comment by kseistrup** [comment]
> https://todo.sr.ht/~sircmpwn/imrsh

That's where I started in the first place. Although I am logged in on sr.ht I don't see any obvious way to add an issue.

The page says

> Tasks for imrsh. Not for end-user support - use the mailing list.

And the mailinglist archive [says](https://lists.sr.ht/~emersion/mrsh-dev/%3CVN-XTs-QaLeiuC-f15QCNIAWtVssX7Fx3bAQztrbPUxULmSfD26qJfF_rg3MyaYm64VnZ37JjkC4CF_-eXNEPYN1E2z_s6HZsKyuexNmCtE%3D%40emersion.fr%3E) that the official issue tracker is the one on Github, which is why I submitted the issue here and put `imrsh` in the subject line.

If I am supposed to join a mailing list in order to submit an issue, I think I'll just pass.

Cheers.
